src/transform/src/lib.rs

```diff
@@ -249,13 +250,16 @@ impl Optimizer {
                     // Predicate pushdown sets the equivalence classes of joins.
                     Box::new(crate::predicate_pushdown::PredicatePushdown),
                     // Lifts the information `!isnull(col)`
-                    Box::new(crate::nonnullable::NonNullable),
+                    // Box::new(crate::nonnullable::NonNullable),
                     // Lifts the information `col = literal`
                     // TODO (#6613): this also tries to lift `!isnull(col)` but
                     // less well than the previous transform. Eliminate
                     // redundancy between the two transforms.
-                    Box::new(crate::column_knowledge::ColumnKnowledge),
-                    // Lifts the information `col1 = col2`
+                    // Box::new(crate::column_knowledge::ColumnKnowledge),
+                    Box::new(crate::predicate_propagation::PredicateKnowledge),
+                    Box::new(crate::reduction_pushdown::ReductionPushdown),
+                    Box::new(crate::redundant_join::RedundantJoin),
+                    Box::new(crate::topk_elision::TopKElision),
                     Box::new(crate::demand::Demand),
```

Comment:
Can `PredicateKnowledge` replace the `Demand` transform? Or at least this instance of it?

Reply:
I've started working on getting predicate knowledge and projection pushdown to team up, but it is a little tricky at the moment. I was hoping to land partial work first to avoid the various rebases. The necessary work includes at least "when you reach a union, record known predicates that you will lose due to the union as filter expressions on the inputs to the union", which .. seems a bit weird but maybe defensible.